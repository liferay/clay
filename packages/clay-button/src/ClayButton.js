import 'clay-icon';
import ClayComponent from 'clay-component';
import defineWebComponent from 'metal-web-component';
import Soy from 'metal-soy';
import {Config} from 'metal-state';

import templates from './ClayButton.soy.js';

/**
 * Metal Clay Button component.
 * @extends ClayComponent
 */
class ClayButton extends ClayComponent {}

/**
 * State definition.
 * @static
 * @type {!Object}
 */
ClayButton.STATE = {
	/**
	 * Aria label attribute for the button element.
	 * @default undefined
	 * @instance
	 * @memberof ClayButton
	 * @type {?(string|undefined)}
	 */
	ariaLabel: Config.string(),

	/**
	 * Renders the button as a block element.
	 * @default false
	 * @instance
	 * @memberof ClayButton
	 * @type {?boolean}
	 */
	block: Config.bool().value(false),

	/**
	 * Data to add to the element.
	 * @default undefined
	 * @instance
	 * @memberof ClayButton
	 * @type {?object}
	 */
	data: Config.object(),

	/**
	 * The button disabled attribute.
	 * @default false
	 * @instance
	 * @memberof ClayButton
	 * @type {?boolean}
	 */
	disabled: Config.bool().value(false),

	/**
	 * CSS classes to be applied to the element.
	 * @default undefined
	 * @instance
	 * @memberof ClayButton
	 * @type {?(string|undefined)}
	 */
	elementClasses: Config.string(),

	/**
	 * Icon to be rendered in the button.
	 * @default undefined
	 * @instance
	 * @memberof ClayButton
	 * @type {?(string|undefined)}
	 */
	icon: Config.string(),

	/**
	 * Render ClayIcon in the ClayButton element.
	 * @default left
	 * @instance
	 * @memberof ClayButton
	 * @type {?string}
	 */
	iconAlignment: Config.oneOf(['left', 'right']).value('left'),

	/**
	 * Id to be applied to the element.
	 * @default undefined
	 * @instance
	 * @memberof ClayButton
	 * @type {?(string|undefined)}
	 */
	id: Config.string(),

	/**
	 * The label of the button content.
	 * @default undefined
	 * @instance
	 * @memberof ClayButton
	 * @type {?(html|string|undefined)}
	 */
	label: Config.any(),

	/**
	 * Flag to indicate if button should be monospaced.
	 * @default false
	 * @instance
	 * @memberof ClayButton
	 * @type {?bool}
	 */
	monospaced: Config.bool().value(false),

	/**
	 * The name attribute value of the element.
	 * @default undefined
	 * @instance
	 * @memberof ClayButton
	 * @type {?(string|undefined)}
	 */
	name: Config.string(),

	/**
	 * The name that will make the CSS class name that will represent the
	 * button size.
	 * @default undefined
	 * @instance
	 * @memberof ClayButton
	 * @type {?(string|undefined)}
	 */
	size: Config.oneOf(['sm']),

	/**
	 * The path to the SVG spritemap file containing the icons.
	 * @default undefined
	 * @instance
	 * @memberof ClayButton
	 * @type {?(string|undefined)}
	 */
	spritemap: Config.string(),

	/**
	 * The css class that colors the button. Style `unstyled` is only for internal
	 * purposes.
	 * @default primary
	 * @instance
	 * @memberof ClayButton
	 * @type {?(bool|string|undefined)}
	 */
	style: Config.oneOfType([
		Config.bool(),
		Config.oneOf([
			'link',
			'primary',
			'secondary',
			'unstyled',
		]),
	]).value('primary'),

	/**
	 * The title attribute of the element.
	 * @default undefined
	 * @instance
	 * @memberof ClayButton
	 * @type {?(string|undefined)}
	 */
	title: Config.string(),

	/**
	 * The type attribute value of the element.
	 * @default button
	 * @instance
	 * @memberof ClayButton
	 * @type {?string}
	 */
	type: Config.oneOf(['button', 'reset', 'submit']).value('button'),

	/**
	 * The value attribute value of the element.
	 * @default undefined
	 * @instance
	 * @memberof ClayButton
	 * @type {?(string|undefined)}
	 */
	value: Config.string(),
};

defineWebComponent('clay-button', ClayButton);

Soy.register(ClayButton, templates);

export {ClayButton};
export default ClayButton;
