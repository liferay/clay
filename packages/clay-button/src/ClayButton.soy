{namespace ClayButton}

/**
 * This renders the component's whole content.
 */
{template .render}
	{@param? ariaLabel: string}
	{@param? block: bool}
	{@param? disabled: bool}
	{@param? elementClasses: string}
	{@param? icon: string}
	{@param? iconAlignment: string}
	{@param? id: string}
	{@param? label: html|string}
	{@param? monospaced: bool}
	{@param? name: string}
	{@param? size: string}
	{@param? spritemap: string}
	{@param? style: bool|string}
	{@param? title: string}
	{@param? type: string}
	{@param? value: string}

	{let $attributes kind="attributes"}
		class="btn
			{if $block}
				{sp}btn-block
			{/if}

			{if $elementClasses}
				{sp}{$elementClasses}
			{/if}

			{if $monospaced}
				{sp}btn-monospaced
			{/if}

			{if $size}
				{sp}btn-{$size}
			{/if}

			{if isNonnull($style) and $style != true and $style != false}
				{sp}btn-{$style}
			{elseif not isNonnull($style) or $style == true}
				{sp}btn-primary
			{/if}
		"

		{if $ariaLabel}
			aria-label="{$ariaLabel}"
		{elseif $icon and not $label}
			aria-label="{$icon}"
		{/if}

		{if $disabled}
			disabled="disabled"
		{/if}

		{if $id}
			id="{$id}"
		{/if}

		{if $name}
			name="{$name}"
		{/if}

		{if $title}
			title="{$title}"
		{/if}

		{if $value}
			value="{$value}"
		{/if}

		type="{$type}"
	{/let}

	<button {$attributes}>
		{call .content}
			{param icon: $icon /}
			{param iconAlignment: $iconAlignment ?: 'left' /}
			{param label: $label /}
			{param spritemap: $spritemap /}
		{/call}
	</button>
{/template}

/**
 * Renders element content
 */
{template .content}
	{@param? icon: string}
	{@param? iconAlignment: string}
	{@param? label: html|string}
	{@param? spritemap: string}

	{let $iconContent kind="html"}
		{if $icon and $spritemap}
			{call .icon}
				{param icon: $icon /}
				{param iconAlignment: $iconAlignment /}
				{param label: $label /}
				{param spritemap: $spritemap /}
			{/call}
		{/if}
	{/let}

	{if $iconContent != '' and $iconAlignment == 'left'}
		{$iconContent}
	{/if}

	{$label ?: ''}

	{if $iconContent != '' and $iconAlignment == 'right'}
		{$iconContent}
	{/if}
{/template}

/**
 * Renders icon element
 */
{template .icon}
	{@param icon: string}
	{@param spritemap: string}
	{@param? iconAlignment: string}
	{@param? label: html|string}

	{let $iconContent kind="html"}
		{call ClayIcon.render}
			{param spritemap: $spritemap /}
			{param symbol: $icon /}
		{/call}
	{/let}

	{if $label}
		{let $iconWrapperClasses kind="text"}
			inline-item
			{if $iconAlignment == 'left'}
				{sp}inline-item-before
			{elseif $iconAlignment == 'right'}
				{sp}inline-item-after
			{/if}
		{/let}

		<span class="{$iconWrapperClasses}">
			{$iconContent}
		</span>
	{else}
		{$iconContent}
	{/if}

{/template}
