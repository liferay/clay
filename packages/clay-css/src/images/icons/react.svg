<!--
* SPDX-FileCopyrightText: © 2020 Liferay, Inc. <https://liferay.com>
* SPDX-FileCopyrightText: © 2020 Contributors to the project Clay <https://github.com/liferay/clay/graphs/contributors>
*
* SPDX-License-Identifier: BSD-3-Clause
-->
<svg viewBox="0 0 16 16" xmlns="http://www.w3.org/2000/svg"><path d="M16 8.006c0-1.041-1.327-2.028-3.362-2.64.47-2.038.26-3.66-.66-4.179a1.438 1.438 0 0 0-.726-.18v-.003c-.897 0-2.071.628-3.259 1.718-1.187-1.083-2.36-1.705-3.258-1.705v.003c-.274 0-.521.058-.737.18-.916.519-1.122 2.137-.649 4.169C1.321 5.984 0 6.968 0 8.006c0 1.041 1.327 2.028 3.362 2.64-.47 2.038-.26 3.66.66 4.179.211.122.459.18.733.18.897 0 2.07-.629 3.258-1.718 1.187 1.083 2.361 1.705 3.258 1.705.274 0 .522-.058.737-.18.917-.519 1.122-2.137.65-4.169C14.678 10.031 16 9.044 16 8.006Zm-4.752-6.287c.15.003.271.032.372.086.444.25.636 1.202.486 2.426a9.73 9.73 0 0 1-.166.942 16.066 16.066 0 0 0-2.071-.349 15.698 15.698 0 0 0-1.357-1.602c1.06-.971 2.058-1.5 2.736-1.503ZM5.11 9.647c.154.256.31.506.47.75a14.93 14.93 0 0 1-1.337-.254c.12-.413.27-.84.44-1.265.134.256.274.512.427.769Zm-.877-3.778c.427-.1.877-.186 1.344-.253a17.649 17.649 0 0 0-.894 1.528 13.57 13.57 0 0 1-.45-1.275Zm.832 2.147a17.028 17.028 0 0 1 1.461-2.503 17.267 17.267 0 0 1 2.942-.003 16.59 16.59 0 0 1 1.468 2.493 17.032 17.032 0 0 1-1.461 2.503 17.251 17.251 0 0 1-2.943.003 16.59 16.59 0 0 1-1.468-2.493Zm5.824 1.621c.15-.256.29-.516.425-.772.176.43.326.859.45 1.275-.428.103-.878.19-1.344.256.16-.246.32-.5.47-.759Zm-.003-3.272a17.37 17.37 0 0 0-.47-.75c.464.068.91.151 1.338.254-.12.413-.27.84-.44 1.265a15.136 15.136 0 0 0-.428-.769Zm-1.99-1.611A20.543 20.543 0 0 0 8 4.73c-.307 0-.61.007-.907.023.294-.375.597-.718.897-1.026.304.308.607.65.907 1.026ZM4.36 1.818a.736.736 0 0 1 .375-.083v-.003c.676 0 1.677.529 2.74 1.493a15.197 15.197 0 0 0-1.347 1.599c-.737.077-1.435.196-2.074.353a9.516 9.516 0 0 1-.17-.93c-.153-1.224.036-2.176.476-2.429Zm-.82 8.136c-.323-.1-.63-.208-.914-.327C1.471 9.144.724 8.51.724 8.006c0-.503.747-1.14 1.901-1.621.28-.119.587-.225.904-.324.186.628.43 1.282.734 1.951-.3.667-.542 1.317-.724 1.942Zm1.21 4.339a.79.79 0 0 1-.37-.086c-.443-.25-.635-1.202-.485-2.426.036-.301.095-.619.166-.942.64.153 1.337.272 2.071.349.44.593.897 1.131 1.357 1.602-1.063.971-2.061 1.503-2.74 1.503Zm3.258-2.01A13.3 13.3 0 0 1 7.1 11.26c.293.012.593.022.897.022.306 0 .61-.007.906-.023-.293.375-.596.718-.896 1.026Zm3.633 1.91a.736.736 0 0 1-.375.084c-.676 0-1.677-.529-2.74-1.493.457-.471.913-1.006 1.347-1.6a15.478 15.478 0 0 0 2.074-.352c.075.324.134.635.17.933.153 1.224-.036 2.175-.476 2.429Zm1.731-4.566c-.28.119-.587.225-.903.324A15.22 15.22 0 0 0 11.734 8c.3-.667.542-1.317.724-1.942.323.1.63.208.917.327 1.154.483 1.901 1.118 1.901 1.621-.003.503-.75 1.14-1.905 1.621Z"/><path d="M7.997 9.47c.823 0 1.49-.655 1.49-1.464 0-.809-.667-1.464-1.49-1.464s-1.49.655-1.49 1.464c0 .809.667 1.464 1.49 1.464Z"/></svg>
