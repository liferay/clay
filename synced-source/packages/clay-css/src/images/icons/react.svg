<!--
* SPDX-FileCopyrightText: © 2020 Liferay, Inc. <https://liferay.com>
* SPDX-FileCopyrightText: © 2020 Contributors to the project Clay <https://github.com/liferay/clay/graphs/contributors>
*
* SPDX-License-Identifier: BSD-3-Clause
-->
<svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 512 512">
	<path class="lexicon-icon-outline" d="M512,259.8c0-33.9-42.5-66-107.6-85.9c15-66.3,8.3-119.1-21.1-136c-6.8-4-14.6-5.8-23.3-5.8V32c-28.7,0-66.3,20.4-104.3,55.9c-38-35.2-75.6-55.5-104.3-55.5v0.1c-8.8,0-16.7,1.9-23.6,5.8C98.6,55.3,92,107.9,107.2,174C42.3,194,0,226.1,0,259.8c0,33.9,42.5,66,107.6,85.9c-15,66.3-8.3,119.1,21.1,136c6.8,4,14.7,5.8,23.5,5.8c28.7,0,66.3-20.4,104.3-55.9c38,35.2,75.6,55.5,104.3,55.5c8.8,0,16.7-1.9,23.6-5.8c29.3-16.9,35.9-69.6,20.8-135.7C469.7,325.7,512,293.6,512,259.8z M359.9,55.3c4.8,0.1,8.7,1,11.9,2.8c14.2,8.1,20.3,39.1,15.5,78.9c-1.1,9.8-3,20.1-5.3,30.7c-20.5-5-42.8-8.9-66.3-11.4c-14.1-19.3-28.7-36.8-43.4-52.1C306.3,72.6,338.2,55.4,359.9,55.3z M163.5,313.2c4.9,8.3,9.9,16.5,15,24.4c-14.8-2.2-29.1-4.9-42.8-8.2c3.9-13.5,8.7-27.3,14.1-41.2C154.1,296.5,158.6,304.9,163.5,313.2z M135.5,190.3c13.7-3.2,28.1-6,43-8.2c-5.1,8-10.2,16.3-15,24.7c-4.8,8.3-9.3,16.7-13.6,25C144.2,217.8,139.4,203.8,135.5,190.3z M162.1,260.2c6.5-14,13.8-27.9,21.6-41.6c8.1-14.1,16.5-27.4,25.2-39.8c15.6-1.4,31.3-2.1,47.2-2.1c15.8,0,31.5,0.7,47,2c8.7,12.4,17.1,25.7,25.3,39.6c7.9,13.7,15.1,27.5,21.7,41.5c-6.6,14-13.8,27.9-21.6,41.6c-8.1,14.1-16.5,27.4-25.2,39.8c-15.5,1.4-31.3,2.1-47.2,2.1c-15.8,0-31.5-0.7-47-2c-8.7-12.4-17.1-25.7-25.3-39.6C175.9,288,168.6,274.1,162.1,260.2z M348.5,312.9c4.8-8.3,9.3-16.8,13.6-25.1c5.6,14,10.4,27.9,14.4,41.5c-13.7,3.3-28.1,6.2-43,8.3C338.5,329.6,343.7,321.4,348.5,312.9z M348.4,206.5c-4.8-8.3-9.9-16.5-15-24.4c14.8,2.2,29.1,4.9,42.8,8.2c-3.9,13.5-8.7,27.3-14.1,41.2C357.8,223.1,353.3,214.8,348.4,206.5z M284.7,154c-9.4-0.4-19-0.7-28.7-0.7c-9.8,0-19.5,0.2-29,0.7c9.4-12.2,19.1-23.4,28.7-33.4C265.4,130.6,275.1,141.8,284.7,154z M139.5,58.5c3.1-1.9,7.2-2.7,12-2.7v-0.1c21.6,0,53.6,17.2,87.7,48.6c-14.6,15.3-29.2,32.7-43.1,52c-23.6,2.5-45.9,6.4-66.4,11.5c-2.4-10.4-4.2-20.5-5.4-30.2C119.4,97.7,125.4,66.7,139.5,58.5z M113.2,323.2c-10.3-3.2-20.1-6.8-29.2-10.6c-36.9-15.7-60.8-36.4-60.8-52.8s23.9-37.1,60.8-52.8c9-3.9,18.8-7.3,28.9-10.5c5.9,20.4,13.8,41.7,23.5,63.5C126.8,281.7,119.1,302.9,113.2,323.2z M152,464.4c-4.7-0.1-8.7-1-11.8-2.8c-14.2-8.1-20.4-39.1-15.6-78.9c1.1-9.8,3-20.1,5.3-30.7c20.5,5,42.8,8.9,66.3,11.4c14.1,19.3,28.7,36.8,43.4,52.1C205.6,447.1,173.7,464.4,152,464.4z M256.2,399c-9.7-10-19.4-21.2-29-33.4c9.4,0.4,19,0.7,28.7,0.7c9.8,0,19.5-0.2,29-0.7C275.5,377.9,265.8,389,256.2,399z M372.5,461.2c-3.1,1.9-7.2,2.7-12,2.7c-21.6,0-53.6-17.2-87.7-48.6c14.6-15.3,29.2-32.7,43.1-52c23.6-2.5,45.9-6.4,66.4-11.5c2.4,10.5,4.3,20.6,5.4,30.3C392.6,422,386.6,453,372.5,461.2z M427.9,312.6c-9,3.9-18.8,7.3-28.9,10.5c-5.9-20.4-13.8-41.7-23.5-63.5c9.6-21.7,17.3-42.9,23.2-63.2c10.3,3.2,20.1,6.8,29.3,10.6c36.9,15.7,60.8,36.4,60.8,52.8C488.7,276.2,464.8,297,427.9,312.6z"/>
	<path class="lexicon-icon-outline" d="M255.9,307.5c26.3,0,47.7-21.3,47.7-47.7s-21.4-47.7-47.7-47.7c-26.3,0-47.7,21.3-47.7,47.7S229.6,307.5,255.9,307.5z"/>
</svg>