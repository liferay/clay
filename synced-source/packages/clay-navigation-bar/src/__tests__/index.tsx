/**
 * SPDX-FileCopyrightText: Â© 2019 Liferay, Inc. <https://liferay.com>
 * SPDX-License-Identifier: BSD-3-Clause
 */

import '@testing-library/jest-dom';
import ClayNavigationBar from '..';
import ClayButton from '@clayui/button';
import ClayLink from '@clayui/link';
import {act, cleanup, fireEvent, render, waitFor} from '@testing-library/react';
import React from 'react';

const spritemap = 'node_modules/clay-css/lib/images/icons/icons.svg';

describe('ClayNavigationBar', () => {
	afterEach(cleanup);

	it('renders', () => {
		const label = 'Item 1';

		const {container} = render(
			<ClayNavigationBar
				inverted
				messages={{
					close: 'Close',
					open: 'Open',
					trigger: '{0} Menu, Current Page: {1}',
				}}
				spritemap={spritemap}
				triggerLabel={label}
			>
				<ClayNavigationBar.Item active>
					<ClayLink href="#1">{label}</ClayLink>
				</ClayNavigationBar.Item>

				<ClayNavigationBar.Item>
					<ClayButton>Item 2</ClayButton>
				</ClayNavigationBar.Item>

				<ClayNavigationBar.Item>
					<ClayLink href="#3">Item 3</ClayLink>
				</ClayNavigationBar.Item>
			</ClayNavigationBar>
		);

		expect(container).toMatchSnapshot();
	});

	it('renders a custom item', () => {
		const label = 'Item 1';

		const {container} = render(
			<ClayNavigationBar
				inverted
				messages={{
					close: 'Close',
					open: 'Open',
					trigger: '{0} Menu, Current Page: {1}',
				}}
				spritemap={spritemap}
				triggerLabel={label}
			>
				<ClayNavigationBar.Item active>
					<a className="my-custom-class" href="#1">
						{label}
					</a>
				</ClayNavigationBar.Item>
			</ClayNavigationBar>
		);

		expect(container).toMatchSnapshot();
	});

	it('renders a dropdown when clicking the trigger element from NavigationBar', async () => {
		jest.useFakeTimers();

		const {container, getByText} = render(
			<ClayNavigationBar
				inverted
				messages={{
					close: 'Close',
					open: 'Open',
					trigger: '{0} Menu, Current Page: {1}',
				}}
				spritemap={spritemap}
				triggerLabel="Trigger Label"
			>
				<ClayNavigationBar.Item active>
					<ClayLink href="#1">Item 1</ClayLink>
				</ClayNavigationBar.Item>

				<ClayNavigationBar.Item>
					<ClayButton>Item 2</ClayButton>
				</ClayNavigationBar.Item>
			</ClayNavigationBar>
		);

		fireEvent.click(getByText('Trigger Label'));

		act(() => {
			jest.runAllTimers();
		});

		let navigationBarDropdown;

		await act(async () => {
			navigationBarDropdown = await waitFor(() =>
				container.querySelector('.navbar-collapse')
			);
		});

		expect(navigationBarDropdown).toBeDefined();

		expect(navigationBarDropdown).toMatchSnapshot();
	});

	it('collapses the previously expanded dropdown when trigger element is clicked', async () => {
		jest.useFakeTimers();

		const {container, getByTestId} = render(
			<ClayNavigationBar
				inverted
				messages={{
					close: 'Close',
					open: 'Open',
					trigger: '{0} Menu, Current Page: {1}',
				}}
				spritemap={spritemap}
				triggerLabel="Trigger Label"
			>
				<ClayNavigationBar.Item active>
					<ClayLink href="#1">Item 1</ClayLink>
				</ClayNavigationBar.Item>

				<ClayNavigationBar.Item>
					<ClayButton>Item 2</ClayButton>
				</ClayNavigationBar.Item>
			</ClayNavigationBar>
		);

		fireEvent.click(getByTestId('navbarToggler'), 'Trigger Label');

		act(() => {
			jest.runAllTimers();
		});

		fireEvent.click(getByTestId('navbarToggler'), 'Trigger Label');

		act(() => {
			jest.runAllTimers();
		});

		let navigationBarDropdown;

		await act(async () => {
			navigationBarDropdown = await waitFor(() =>
				container.querySelector('.navbar-collapse.collapse')
			);
		});

		expect(navigationBarDropdown).toBeDefined();

		expect(navigationBarDropdown).not.toHaveClass('show');

		expect(navigationBarDropdown).toMatchSnapshot();
	});

	it('renders when passing more than one active item', () => {
		const label = 'Item 1';

		const {container} = render(
			<ClayNavigationBar
				inverted
				messages={{
					close: 'Close',
					open: 'Open',
					trigger: '{0} Menu, Current Page: {1}',
				}}
				spritemap={spritemap}
				triggerLabel={label}
			>
				<ClayNavigationBar.Item>
					<ClayLink href="#1">{label}</ClayLink>
				</ClayNavigationBar.Item>

				<ClayNavigationBar.Item active>
					<ClayButton>Item 2</ClayButton>
				</ClayNavigationBar.Item>

				<ClayNavigationBar.Item>
					<ClayLink href="#3">Item 3</ClayLink>
				</ClayNavigationBar.Item>
			</ClayNavigationBar>
		);

		expect(container).toMatchSnapshot();
	});

	it('throws a warning when passing more than one active prop to child', () => {
		const mockWarnings = jest
			.spyOn(global.console, 'error')
			.mockImplementation(() => null);

		const label = 'Item 1';

		const {container} = render(
			<ClayNavigationBar
				inverted
				messages={{
					close: 'Close',
					open: 'Open',
					trigger: '{0} Menu, Current Page: {1}',
				}}
				spritemap={spritemap}
				triggerLabel={label}
			>
				<ClayNavigationBar.Item active>
					<ClayLink href="#1">{label}</ClayLink>
				</ClayNavigationBar.Item>

				<ClayNavigationBar.Item active>
					<ClayButton>Item 2</ClayButton>
				</ClayNavigationBar.Item>
			</ClayNavigationBar>
		);

		expect(mockWarnings).toBeCalled();
		expect(mockWarnings.mock.calls[0]![0]).toBe(
			'Warning: ClayNavigationBar expects 0 or 1 active children, but received 2'
		);
		expect(container).toMatchSnapshot();
		jest.resetAllMocks();
	});

	it('renders with a single item', () => {
		const label = 'Item 1';

		const {container} = render(
			<ClayNavigationBar
				inverted
				messages={{
					close: 'Close',
					open: 'Open',
					trigger: '{0} Menu, Current Page: {1}',
				}}
				spritemap={spritemap}
				triggerLabel={label}
			>
				<ClayNavigationBar.Item active>
					<ClayLink href="#1">{label}</ClayLink>
				</ClayNavigationBar.Item>
			</ClayNavigationBar>
		);

		expect(container).toMatchSnapshot();
	});
});
