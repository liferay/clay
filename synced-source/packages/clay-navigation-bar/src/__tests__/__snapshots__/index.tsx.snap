// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ClayNavigationBar collapses the previously expanded dropdown when trigger element is clicked 1`] = `
<div
  class="navbar-collapse collapse"
  style="height: auto"
>
  <div
    class="container-fluid container-fluid-max-xl"
  >
    <ul
      class="navbar-nav"
    >
      <li
        class="nav-item"
      >
        <a
          aria-current="page"
          class="nav-link active"
          href="#1"
        >
          <span>
            Item 1
          </span>
        </a>
      </li>
      <li
        class="nav-item"
      >
        <button
          class="nav-link btn"
          type="button"
        >
          <span>
            Item 2
          </span>
        </button>
      </li>
    </ul>
  </div>
</div>
`;

exports[`ClayNavigationBar renders 1`] = `
<div>
  <nav
    class="navbar navbar-collapse-relative navbar-expand-md navbar-underline navigation-bar navigation-bar-secondary"
  >
    <div
      class="container-fluid container-fluid-max-xl"
    >
      <button
        aria-expanded="false"
        aria-label="Open Menu, Current Page: Item 1"
        class="navbar-toggler navbar-toggler-link collapsed btn btn-unstyled"
        data-testid="navbarToggler"
        type="button"
      >
        <span
          class="navbar-text-truncate"
        >
          Item 1
        </span>
        <svg
          class="lexicon-icon lexicon-icon-caret-bottom"
          role="presentation"
        >
          <use
            href="node_modules/clay-css/lib/images/icons/icons.svg#caret-bottom"
          />
        </svg>
      </button>
      <div
        class="navbar-collapse collapse"
      >
        <div
          class="container-fluid container-fluid-max-xl"
        >
          <ul
            class="navbar-nav"
          >
            <li
              class="nav-item"
            >
              <a
                aria-current="page"
                class="nav-link active"
                href="#1"
              >
                <span>
                  Item 1
                </span>
              </a>
            </li>
            <li
              class="nav-item"
            >
              <button
                class="nav-link btn"
                type="button"
              >
                <span>
                  Item 2
                </span>
              </button>
            </li>
            <li
              class="nav-item"
            >
              <a
                class="nav-link"
                href="#3"
              >
                <span>
                  Item 3
                </span>
              </a>
            </li>
          </ul>
        </div>
      </div>
    </div>
  </nav>
</div>
`;

exports[`ClayNavigationBar renders a custom item 1`] = `
<div>
  <nav
    class="navbar navbar-collapse-relative navbar-expand-md navbar-underline navigation-bar navigation-bar-secondary"
  >
    <div
      class="container-fluid container-fluid-max-xl"
    >
      <button
        aria-expanded="false"
        aria-label="Open Menu, Current Page: Item 1"
        class="navbar-toggler navbar-toggler-link collapsed btn btn-unstyled"
        data-testid="navbarToggler"
        type="button"
      >
        <span
          class="navbar-text-truncate"
        >
          Item 1
        </span>
        <svg
          class="lexicon-icon lexicon-icon-caret-bottom"
          role="presentation"
        >
          <use
            href="node_modules/clay-css/lib/images/icons/icons.svg#caret-bottom"
          />
        </svg>
      </button>
      <div
        class="navbar-collapse collapse"
      >
        <div
          class="container-fluid container-fluid-max-xl"
        >
          <ul
            class="navbar-nav"
          >
            <li
              class="nav-item"
            >
              <a
                class="my-custom-class active"
                href="#1"
              >
                Item 1
              </a>
            </li>
          </ul>
        </div>
      </div>
    </div>
  </nav>
</div>
`;

exports[`ClayNavigationBar renders a dropdown when clicking the trigger element from NavigationBar 1`] = `
<div
  class="navbar-collapse show"
  style="height: auto"
>
  <div
    class="container-fluid container-fluid-max-xl"
  >
    <ul
      class="navbar-nav"
    >
      <li
        class="nav-item"
      >
        <a
          aria-current="page"
          class="nav-link active"
          href="#1"
        >
          <span>
            Item 1
          </span>
        </a>
      </li>
      <li
        class="nav-item"
      >
        <button
          class="nav-link btn"
          type="button"
        >
          <span>
            Item 2
          </span>
        </button>
      </li>
    </ul>
  </div>
</div>
`;

exports[`ClayNavigationBar renders when passing more than one active item 1`] = `
<div>
  <nav
    class="navbar navbar-collapse-relative navbar-expand-md navbar-underline navigation-bar navigation-bar-secondary"
  >
    <div
      class="container-fluid container-fluid-max-xl"
    >
      <button
        aria-expanded="false"
        aria-label="Open Menu, Current Page: Item 1"
        class="navbar-toggler navbar-toggler-link collapsed btn btn-unstyled"
        data-testid="navbarToggler"
        type="button"
      >
        <span
          class="navbar-text-truncate"
        >
          Item 1
        </span>
        <svg
          class="lexicon-icon lexicon-icon-caret-bottom"
          role="presentation"
        >
          <use
            href="node_modules/clay-css/lib/images/icons/icons.svg#caret-bottom"
          />
        </svg>
      </button>
      <div
        class="navbar-collapse collapse"
      >
        <div
          class="container-fluid container-fluid-max-xl"
        >
          <ul
            class="navbar-nav"
          >
            <li
              class="nav-item"
            >
              <a
                class="nav-link"
                href="#1"
              >
                <span>
                  Item 1
                </span>
              </a>
            </li>
            <li
              class="nav-item"
            >
              <button
                aria-current="page"
                class="nav-link active btn"
                type="button"
              >
                <span>
                  Item 2
                </span>
              </button>
            </li>
            <li
              class="nav-item"
            >
              <a
                class="nav-link"
                href="#3"
              >
                <span>
                  Item 3
                </span>
              </a>
            </li>
          </ul>
        </div>
      </div>
    </div>
  </nav>
</div>
`;

exports[`ClayNavigationBar renders with a single item 1`] = `
<div>
  <nav
    class="navbar navbar-collapse-relative navbar-expand-md navbar-underline navigation-bar navigation-bar-secondary"
  >
    <div
      class="container-fluid container-fluid-max-xl"
    >
      <button
        aria-expanded="false"
        aria-label="Open Menu, Current Page: Item 1"
        class="navbar-toggler navbar-toggler-link collapsed btn btn-unstyled"
        data-testid="navbarToggler"
        type="button"
      >
        <span
          class="navbar-text-truncate"
        >
          Item 1
        </span>
        <svg
          class="lexicon-icon lexicon-icon-caret-bottom"
          role="presentation"
        >
          <use
            href="node_modules/clay-css/lib/images/icons/icons.svg#caret-bottom"
          />
        </svg>
      </button>
      <div
        class="navbar-collapse collapse"
      >
        <div
          class="container-fluid container-fluid-max-xl"
        >
          <ul
            class="navbar-nav"
          >
            <li
              class="nav-item"
            >
              <a
                aria-current="page"
                class="nav-link active"
                href="#1"
              >
                <span>
                  Item 1
                </span>
              </a>
            </li>
          </ul>
        </div>
      </div>
    </div>
  </nav>
</div>
`;

exports[`ClayNavigationBar throws a warning when passing more than one active prop to child 1`] = `
<div>
  <nav
    class="navbar navbar-collapse-relative navbar-expand-md navbar-underline navigation-bar navigation-bar-secondary"
  >
    <div
      class="container-fluid container-fluid-max-xl"
    >
      <button
        aria-expanded="false"
        aria-label="Open Menu, Current Page: Item 1"
        class="navbar-toggler navbar-toggler-link collapsed btn btn-unstyled"
        data-testid="navbarToggler"
        type="button"
      >
        <span
          class="navbar-text-truncate"
        >
          Item 1
        </span>
        <svg
          class="lexicon-icon lexicon-icon-caret-bottom"
          role="presentation"
        >
          <use
            href="node_modules/clay-css/lib/images/icons/icons.svg#caret-bottom"
          />
        </svg>
      </button>
      <div
        class="navbar-collapse collapse"
      >
        <div
          class="container-fluid container-fluid-max-xl"
        >
          <ul
            class="navbar-nav"
          >
            <li
              class="nav-item"
            >
              <a
                aria-current="page"
                class="nav-link active"
                href="#1"
              >
                <span>
                  Item 1
                </span>
              </a>
            </li>
            <li
              class="nav-item"
            >
              <button
                aria-current="page"
                class="nav-link active btn"
                type="button"
              >
                <span>
                  Item 2
                </span>
              </button>
            </li>
          </ul>
        </div>
      </div>
    </div>
  </nav>
</div>
`;
